core/src/main/java/org/fao/geonet/kernel/security/openidconnect/bearer/MSGraphUserRolesResolver.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2022 Food and Agriculture Organization of the
+ * United Nations (FAO-UN), United Nations World Food Programme (WFP)
+ * and United Nations Environment Programme (UNEP)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or (at
+ * your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ * Contact: Jeroen Ticheler - FAO - Viale delle Terme di Caracalla 2,
+ * Rome - Italy. email: geonetwork@osgeo.org
+ */
+package org.fao.geonet.kernel.security.openidconnect.bearer;
+
+import org.fao.geonet.kernel.security.openidconnect.OIDCConfiguration;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.oauth2.core.oidc.OidcUserInfo;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * make sure your Azure AD application has "GroupMember.Read.All" permission:
+ * a) go to your application in Azure AD (in the portal)
+ * b) On the left, go to "API permissions"
+ * c) click "Add a permission"
+ * d) press "Microsoft Graph"
+ * e) press "Delegated permission"
+ * f) Scroll down to "GroupMember"
+ * g) Choose "GroupMemeber.Read.All"
+ * h) press "Add permission"
+ * i) on the API Permission screen, press the "Grant admin consent for ..." text
+ * <p>
+ * This class will go to the "https://graph.microsoft.com/v1.0/me/memberOf" and attach your access token.
+ * It will then read the response and find all the user's groups.
+ * <p>
+ * NOTE: to be consistent with the rest of Azure, we use the Groups OID (guid) NOT its name.
+ */
+public class MSGraphUserRolesResolver implements UserRolesResolver {
```

Comment:
This class seems not using the internet proxy configuration if a proxy server is configured, to review.

Reply:
hmmm - I don't think the internal spring library is using the proxy either. 

I don't think I can support that right now - and there's insufficient examples in the code base.

will not change.

Reply:
@josegar74  - no action on this.  Not sure what to do...